core/services/ocr2/plugins/functions/reporting.go

```diff
@@ -132,8 +149,21 @@ func (r *functionsReporting) Query(ctx context.Context, ts types.ReportTimestamp
 
 	queryProto := encoding.Query{}
 	var idStrs []string
+	var reportCoordinator *common.Address
 	for _, result := range results {
 		result := result
+		if r.contractVersion == 1 {
+			reportCoordinator, err = ShouldIncludeCoordinator(result.CoordinatorContractAddress, reportCoordinator)
+			if err != nil {
+				r.logger.Debug("FunctionsReporting Query: skipping request with mismatched coordinator contract address", commontypes.LogFields{
+					"requestID":          formatRequestId(result.RequestID[:]),
+					"requestCoordinator": result.CoordinatorContractAddress,
+					"reportCoordinator":  reportCoordinator,
+					"error":              err,
+				})
+				continue
```

Comment:
Since we exit the loop the first time we encounter a request with a different coordinator address, what happens if nodes don't have the results in the same order?

Reply:
IIUC, this shouldn't be a problem as only the leader is calling query, so the leader decides which which coordinator contract is going to be used for the current round.

Reply:
Yes, only the leader calls query in a given round.